app.js

```diff
@@ -1,138 +1,188 @@
+          if (
+            i != res.length - 1 &&
+            (res[i + 1] == "snapshot" || res[i + 1] == "snapshot.")
+          ) {
+            console.log("detected take snapshot");
+            take_snapshot = true;
+          }
         }
-        return {"text":global_text,"flag":take_snapshot};
+      }
+      global_text += text;
+      global_text += " ";
+      // console.log(text);
+      // div.textContent += text;
+      spans[i].classList.add("from-cs");
+    }
+  }
+  return { text: global_text, flag: take_snapshot };
 }
 
-function start_fetching_meet(){ 
-  console.log('started');
+function start_fetching_meet() {
+  console.log("started");
   // while(true){
-      if(catch_meet){
-          //console.log('inside');
-          setTimeout(() => {
-              chrome.tabs.executeScript(null, { code: '(' + get_data + ')();' }, (results) => {
-                  // console.log('FUN RES');
-                  // console.log(results);
-                  // div.innerText += " ";
-                  // div.innerText += results;
-                  google_meet_transcript_data += " ";
-                  google_meet_transcript_data += results[0].text;
-                  console.log(results[0].flag, results,  "value of take snapshot bool");
-                  if(results[0].flag){
-                    console.log('take snapshot!! :)');
-                    takeScreenShot();
-                    acknowledgeSSSuccess();
-                  }
-                  start_fetching_meet();
-              });
-          },5000);
-      }else return;
+  if (catch_meet) {
+    //console.log('inside');
+    setTimeout(() => {
+      chrome.tabs.executeScript(
+        null,
+        { code: "(" + get_data + ")();" },
+        (results) => {
+          // console.log('FUN RES');
+          // console.log(results);
+          // div.innerText += " ";
+          // div.innerText += results;
+          google_meet_transcript_data += " ";
+          google_meet_transcript_data += results[0].text;
+          localStorage.setItem("meet-data", google_meet_transcript_data);
+          console.log(results[0].flag, results, "value of take snapshot bool");
+          if (results[0].flag) {
+            console.log("take snapshot!! :)");
+            takeScreenShot();
+            acknowledgeSSSuccess();
+          }
+          start_fetching_meet();
+        }
+      );
+    }, 5000);
+  } else return;
   // }
 }
 
-function get_host_name(){
+function get_host_name() {
   return window.location.host;
 }
 
-start_button.addEventListener("click",async function(){
-  localStorage.setItem('images',JSON.stringify([]));
-  await chrome.tabs.executeScript(null, { code: '(' + get_host_name + ')();' }, (results) => {
-    console.log(results);
-      if(results[0] == "meet.google.com"){
+start_button.addEventListener("click", async function () {
+  localStorage.setItem("images", JSON.stringify([]));
+  await chrome.tabs.executeScript(
+    null,
```

Comment:
As we are storing the meet data in the localstorage, we need to clean the localstorage too before beginning to store the data.

make localstorage.setitem('google-meet-data','');